### DISCUSSION TITLE
using oceananigans for nearshore hydrodynamics

### DISCUSSION THREAD

**Discussion by tarankalra1** [CliMA/Oceananigans.jl#2166] [Ideas]
Hi guys, 

I am currently exploring non hydrostatic models that can resolve waves in nearshore regions.  The shoaling of nearshore waves causes sediment transport and that is something that we want to study in details. 

Do you guys think that this model can handle such an application ? 

Thank you for the feedback.

**Comment by glwagner** [comment]
Hi @tarankalra-usgs ! If I understand your question correctly, you are looking for a modeling tool that can simulate the non-hydrostatic Navier-Stokes equations with a free surface and variable depth?

Oceananigans `NonhydrostaticModel` requires a rigid lid at the top (or periodic z-direction), so unfortunately we cannot simulate nonhydrostatic free surface flows. Support for complex domains (eg, variable depth) with `NonhydrostaticModel` is also experimental at this time.

You might consider Basilisk: http://basilisk.fr/, though there may be better tools if you need consider only wave _shoaling_ and not wave _breaking_.

**Reply by tarankalra1** [reply]
Thank you for your quick response. Yes, indeed a free surface and variable depth model is what I need. I am also looking at CROCO https://www.croco-ocean.org/ for the application.

**Reply by glwagner** [reply]
Ah yes, CROCO looks like a better choice for applications without explicitly resolved wave breaking!

**Reply by ChitYanToe** [reply]
Hi @glwagner 
"Oceananigans NonhydrostaticModel requires a rigid lid at the top (or periodic z-direction), so unfortunately we cannot simulate nonhydrostatic free surface flows. Support for complex domains (eg, variable depth) with NonhydrostaticModel is also experimental at this time." remains the same yet, right? (just for clarification). 
Thanks a lot and Kind Regards,
Chit

**Reply by glwagner** [reply]
No, that is not correct (where is that written?)

`NonhydrostaticModel` supports an implicit free surface. Also I think that support for complex domains is fairly mature at this point.

**Reply by ChitYanToe** [reply]
I just saw the reply.

> Hi @tarankalra-usgs ! If I understand your question correctly, you are looking for a modeling tool that can simulate the non-hydrostatic Navier-Stokes equations with a free surface and variable depth?
> 
> Oceananigans `NonhydrostaticModel` requires a rigid lid at the top (or periodic z-direction), so unfortunately we cannot simulate nonhydrostatic free surface flows. Support for complex domains (eg, variable depth) with `NonhydrostaticModel` is also experimental at this time.
> 
> You might consider Basilisk: http://basilisk.fr/, though there may be better tools if you need consider only wave _shoaling_ and not wave _breaking_.

**Reply by glwagner** [reply]
I noticed that I was the one who said that! 😆 Sorry --- what I said back in 2022 is no longer true!
